tests/run/carray_slicing.pyx

```diff
@@ -275,3 +276,227 @@ def struct_ptr_iter():
     return ([ value.i for value in my_structs[:5] ],
             [ ptr.i for ptr in my_structs[:5] ],
             [ inferred.i for inferred in my_structs[:5] ])
+
+############################################################
+# tests for coercing slice of c-array to python list
+
+def test_coercing_c_slice_to_py_start_end():
+    """
+    >>> test_coercing_c_slice_to_py_start_end()
+    [4, 9]
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[2:4]
+
+def test_coercing_c_slice_to_py__end():
+    """
+    >>> test_coercing_c_slice_to_py__end()
+    [0, 1]
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[:2]
+
+
+def test_coercing_c_slice_to_py_from_():
+    """
+    >>> test_coercing_c_slice_to_py_from_()
+    [64, 81]
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[8:]
+
+cpdef list test_coercing_c_slice_in_cpedef_list():
+    """
+    >>> test_coercing_c_slice_in_cpedef_list()
+    [64, 81]
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[8:]
+
+
+def test_c_to_py_coercing_array_in_var():
+    """
+    >>> test_c_to_py_coercing_array_in_var()
+    [64, 81]
+    """
+    cdef int i
+    cdef int a[10]
+    b = 5555
+    for i in range(10):
+        a[i] = i**2
+    b = a[8:]
+    return b
+
+cpdef list test_c_to_py_coercing_array_in_var_2():
```

Comment:
I don't see why specifying the return type here makes any difference to what the test does?

Reply:
ok, will change